### PR TITLE
Gmean, Hmean expanded for other datatypes

### PR DESCRIPTION
**Pull request by Madhav2310**

### CODE REVIEW

File: src/runtime/statistics.py [hunk 1]
```diff
@@ -134,6 +173,48 @@ def harmonic_mean(x: list[i32]) -> f64:
     for i in range(k):
         if x[i] == 0:
             return 0.0
+        if(x[i]<0.0):
+            raise Exception("Harmonic mean does not support negative values")
+        sum += 1 / x[i]
+
+    return float(k/sum)
+@overload
+def harmonic_mean(x: list[i64]) -> f64:
```
Comment (Smit-create) [inline src/runtime/statistics.py:182]:
```suggestion
    return float(k/sum)
 
@overload
def harmonic_mean(x: list[i64]) -> f64:
```

File: src/runtime/statistics.py [hunk 2]
```diff
@@ -115,11 +115,50 @@ def geometric_mean(x: list[i32]) -> f64:
...
+    product = 1.0
+    i: i32
+
+    for i in range(k):
+        if(x[i]<=0.0):
+            raise Exception("geometric mean requires a non-empty dataset containing positive numbers")
+        product *= x[i]
+
+    return product**(1/k)
+@overload
```
Comment (Smit-create) [inline src/runtime/statistics.py:161]:
```suggestion
    return product**(1/k)

@overload
```

File: src/runtime/statistics.py [hunk 3]
```diff
@@ -115,11 +115,51 @@ def geometric_mean(x: list[i32]) -> f64:
     i: i32
 
     for i in range(k):
+        if(x[i]<=0):
```
Comment (czgdp1807) [inline src/runtime/statistics.py:118]:
```suggestion
        if x[i] <= 0:
```

File: src/runtime/statistics.py [hunk 4]
```diff
@@ -115,11 +115,51 @@ def geometric_mean(x: list[i32]) -> f64:
...
+    """
+    k: i32 = len(x)
+    if k == 0:
+        return 0.0
+    product: f64
+    product = 1.0
+    i: i32
+
+    for i in range(k):
+        if(x[i]<=0):
```
Comment (czgdp1807) [inline src/runtime/statistics.py:137]:
```suggestion
        if x[i] <= 0:
```

File: src/runtime/statistics.py [hunk 5]
```diff
@@ -115,11 +115,51 @@ def geometric_mean(x: list[i32]) -> f64:
...
+    """
+    k: i32 = len(x)
+    if k == 0:
+        return 0.0
+    product: f64
+    product = 1.0
+    i: i32
+
+    for i in range(k):
+        if(x[i]<=0.0):
```
Comment (czgdp1807) [inline src/runtime/statistics.py:156]:
```suggestion
        if x[i] <= 0.0:
```

File: src/runtime/statistics.py [hunk 6]
```diff
@@ -134,6 +174,49 @@ def harmonic_mean(x: list[i32]) -> f64:
     for i in range(k):
         if x[i] == 0:
             return 0.0
+        if(x[i]<0.0):
```
Comment (czgdp1807) [inline src/runtime/statistics.py:177]:
```suggestion
        if x[i] < 0.0:
```

File: src/runtime/statistics.py [hunk 7]
```diff
@@ -134,6 +174,49 @@ def harmonic_mean(x: list[i32]) -> f64:
...
+    if k == 0:
+        return 0.0
+    sum: f64
+    sum = 0.0
+    i: i32
+
+    for i in range(k):
+        if x[i] == 0:
+            return 0.0
+        if(x[i]<0):
```
Comment (czgdp1807) [inline src/runtime/statistics.py:198]:
```suggestion
        if x[i] < 0:
```

File: src/runtime/statistics.py [hunk 8]
```diff
@@ -134,6 +174,49 @@ def harmonic_mean(x: list[i32]) -> f64:
...
+    if k == 0:
+        return 0.0
+    sum: f64
+    sum = 0.0
+    i: i32
+
+    for i in range(k):
+        if x[i] == 0.0:
+            return 0.0
+        if(x[i]<0.0):
```
Comment (czgdp1807) [inline src/runtime/statistics.py:218]:
```suggestion
        if x[i] < 0.0:
```

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

**Comment by namannimmo10** [comment]
@Madhav2310 -- please read [PEP 8](https://peps.python.org/pep-0008/).

File: integration_tests/test_statistics.py [hunk 9]
```diff
@@ -45,13 +45,31 @@ def test_geometric_mean():
...
+
 def test_harmonic_mean():
     c: list[i32]
     c = [9,2,46]
     k: f64
     k = harmonic_mean(c)
     assert abs(k - 4.740458015267175) < eps
 
+    d: list[i32]
+    d = [9,0,46]
```
Comment (czgdp1807) [inline integration_tests/test_statistics.py:62]:
```suggestion
    d = [9, 0, 46]
```

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

**Review by czgdp1807 — APPROVED** [review]

**Comment by czgdp1807** [comment]
I removed `coviariance` and `correlation` for now from this PR. You can add it in a new PR. Thanks for contributing.

**Pull request merged**